libs/application/ui-shell/src/reducer/reducerUtils.spec.ts

```diff
@@ -45,27 +47,34 @@ describe('reducerUtils', () => {
       sectionIndex,
       subSectionIndex,
     })
+
     const screens: FormScreen[] = [
       buildIntroScreen('intro'),
       buildTextScreen('a'),
       buildTextScreen('b'),
       buildTextScreen('c'),
     ]
+
     it('should default to the first screen if there are no answers', () => {
       expect(findCurrentScreen(screens, {})).toBe(0)
     })
+
     it('should default to the first screen if the answers dont really match the list of screens', () => {
       expect(findCurrentScreen(screens, { random: 'asdf', notThis: '4' })).toBe(
         0,
       )
     })
+
     it('should go to the screen where the last answer belongs to the screen before', () => {
       expect(findCurrentScreen(screens, { a: 'answer' })).toBe(2)
-      expect(findCurrentScreen(screens, { b: 'answer' })).toBe(3)
+      expect(findCurrentScreen(screens, { b: 'answer' })).toBe(0)
+      expect(findCurrentScreen(screens, { a: 'answer', b: 'answer' })).toBe(3)
     })
-    it('should go to the last screen if the last question has an answer', () => {
```

Comment:
I believe that shouldn't happen. If an answer is missing in between the answers, it should go the first missing answer. That's the purpose of the changes done here

Reply:
I agree 💯